sources/packages/backend/libs/integrations/src/esdc-integration/student-loan-balances/student-loan-balances.processing.service.ts

```diff
@@ -0,0 +1,136 @@
+        new RegExp(
+          `^${this.esdcConfig.environmentCode}EDU.PBC.PT.MBAL.[0-9]{8}.*`,
+          "i",
+        ),
+      );
+    for (const remoteFilePath of remoteFilePaths) {
+      await this.processFile(remoteFilePath, serviceProcessSummary);
+      parentProcessSummary.children(serviceProcessSummary);
+    }
+  }
+
+  /**
+   * Process each individual Student Loan Balances response file from the SFTP.
+   * @param remoteFilePath Student Loan Balances response file to be processed.
+   * @returns process summary and errors summary.
+   */
+  private async processFile(
+    remoteFilePath: string,
+    childrenProcessSummary: ProcessSummary,
+  ): Promise<void> {
+    childrenProcessSummary.info(`Processing file ${remoteFilePath}.`);
+    let studentLoanBalancesSFTPResponseFile: StudentLoanBalancesSFTPResponseFile;
+    try {
+      studentLoanBalancesSFTPResponseFile =
+        await this.studentLoanBalancesIntegrationService.downloadResponseFile(
+          remoteFilePath,
+        );
+    } catch (error) {
+      this.logger.error(error);
+      childrenProcessSummary.error(
+        `Error downloading file ${remoteFilePath}. Error: ${error}`,
+      );
+    }
+    childrenProcessSummary.info(
+      `File contains ${studentLoanBalancesSFTPResponseFile.records.length} Student Loan balances.`,
+    );
+    // Get only the file name for logging.
+    const fileName = path.basename(remoteFilePath);
+    try {
+      await this.dataSource.transaction(async (transactionalEntityManager) => {
+        const studentLoanBalancesRepo =
+          transactionalEntityManager.getRepository(StudentLoanBalances);
+        for (const studentLoanBalanceRecord of studentLoanBalancesSFTPResponseFile.records) {
+          const student = await this.studentService.getStudentByPersonalInfo(
+            studentLoanBalanceRecord.sin,
+            studentLoanBalanceRecord.lastName,
+            getISODateOnlyString(studentLoanBalanceRecord.birthDate),
+            transactionalEntityManager,
+          );
+          // If student not found continue.
+          if (!student) {
+            childrenProcessSummary.error(
+              `Student not found for line ${studentLoanBalanceRecord.lineNumber}.`,
+            );
+            continue;
+          }
+
+          await studentLoanBalancesRepo.insert({
+            studentId: student.id,
+            cslBalance: studentLoanBalanceRecord.cslBalance,
+            balanceDate: getISODateOnlyString(
+              studentLoanBalancesSFTPResponseFile.header.balanceDate,
+            ),
+          });
+        }
+      });
+      childrenProcessSummary.info(
+        `Inserted Student Loan balances file  ${fileName}.`,
+      );
+    } catch (error) {
+      // Log the error but allow the process to continue.
+      const errorDescription = `Error processing file ${fileName}.`;
+      childrenProcessSummary.error(errorDescription);
+      this.logger.error(`${errorDescription}. ${error}`);
+    }
+    try {
```

Comment:
The file removal as it is right now will delete the file even if the file had issues, right?
Should we delete the file only if the file was successfully imported?

Reply:
Nice question, i was in an understanding that it should proceed futher by deleting it.

Reply:
What is the conclusion here? Are we going to delete the file even if there was an issue with some record?

Reply:
We have other services following the same and we do not have a clear direction on this. Unless @JasonCTang has something different to point.
We also have a ticket in place to archive files instead of deleting them.